tests/YearMonthTest.php

```diff
@@ -330,16 +330,38 @@ public function testWithYear()
         $this->assertYearMonthIs(2001, 5, YearMonth::of(2000, 5)->withYear(2001));
     }
 
+    public function testWithYearIsTheSameYear()
+    {
+        $year = 2018;
+        $month = 2;
+        $yearMonth = YearMonth::of($year, $month);
+
+        $this->assertInstanceOf(YearMonth::class, $yearMonth->withYear($year));
```

Comment:
This line is not needed, assertYearMonthIs() will throw a TypeError if it's not an instance of YearMonth.